### PR TITLE
Add close button to in-article email sign-up

### PR DESCRIPTION
**Pull request by gtrufitt**
## What does this change?

This allows users to remove the email sign-up from in an article. It stores this preference in localstorage so that the user does not see the same email sign-up on the same machine.
## What is the value of this and can you measure success?

This improves the user experience of in-article sign-up forms regardless of the position of this insertion putting the decision to remove in the user's hands (and remembering that decision).
## Does this affect other platforms - Amp, Apps, etc?

Nope
## Screenshots

image

### CODE REVIEW

File: static/src/stylesheets/module/email/_form.scss [hunk 1]
```diff
@@ -163,6 +163,28 @@
...
+    top: 0;
+    right: 0;
+}
+
+.email-sub--hidden {
+    @include u-h();
+}
+
+.email-sub--close-icon svg {
+    fill: darken(guss-colour(neutral-2, $pasteup-palette), 10%);
```
Comment (sndrs) [inline static/src/stylesheets/module/email/_form.scss]: this can just be `colour(neutral-2)`.

guss is a bit crazy, it's something we'll tidy up

Comment (gtrufitt) [inline static/src/stylesheets/module/email/_form.scss]: Ah yeah, cool.

File: static/src/javascripts/projects/common/modules/email/email-article.js [hunk 2]
```diff
@@ -138,10 +140,17 @@ define([
         },
         listCanRun = function (listConfig) {
             // Check our lists canRun method and make sure that the user isn't already subscribed to this email
-            if (listConfig.canRun && canRunList[listConfig.canRun]() && !contains(userListSubscriptions, listConfig.listId)) {
+            if (listConfig.canRun &&
```
Comment (johnduffell) [inline static/src/javascripts/projects/common/modules/email/email-article.js]: canRun is a misleading name (I know it's existing code) but I think this is more like listName?

**Comment by johnduffell** [comment]
the cross seems quite small, is it big enough on mobile to target with fat fingers?

**Comment by johnduffell** [comment]
awesome idea, power to the users - they love it! :+1:

**Comment by gtrufitt** [comment]
> the cross seems quite small, is it big enough on mobile to target with fat fingers?

Yeah, good shout, it has padding around it though so the click area is nice and big :+1: 

> power to the users

:muscle:

File: static/src/javascripts/projects/common/modules/email/email-article.js (hunk 2 above)
Comment (gtrufitt) [inline static/src/javascripts/projects/common/modules/email/email-article.js]: Good point, I'll update this.

**Comment by paperboyo** [comment]
I love it. The smallest problem possible: in beta and release Firefoxes on both Win and Mac, the close button has no hover state. Works well in Chrome:

no hand

**Pull request merged**